### PR TITLE
Feat: Migrate navigation components from `./mobile` to `@dzcode.io/ui-mobile`

### LINKED ISSUE #499: migrate navigation components from `./mobile` to `@dzcode.io/ui-mobile`

### PR DESCRIPTION
**Pull request by omdxp**
# Description

This PR is dedicated to migrate all navigation components from `./mobile` to `@dzcode.io/ui-mobile`

Resolves #499 

## Type of change

- New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: packages/ui-mobile/package.json [hunk 1]
```diff
@@ -8,8 +8,12 @@
...
   },
   "dependencies": {
-    "react-native-paper": "^4.9.2",
-    "react-native-vector-icons": "^8.1.0"
+    "@react-navigation/bottom-tabs": "5.11.2",
+    "@react-navigation/drawer": "^5.12.5",
+    "@react-navigation/native": "~5.8.10",
+    "@react-navigation/stack": "~5.12.8",
+    "react-native-markdown-display": "^7.0.0-alpha.2",
+    "react-native-paper": "^4.9.2"
```
Comment (ZibanPirate) [inline packages/ui-mobile/package.json:16]: let's then remove these:

https://github.com/dzcode-io/dzcode.io/blob/030bbbb36ea36151f49d0cc0205bd28cc5e90ea0/mobile/package.json#L18-L21

and these:

https://github.com/dzcode-io/dzcode.io/blob/030bbbb36ea36151f49d0cc0205bd28cc5e90ea0/mobile/package.json#L40-L41

**Review by ZibanPirate — CHANGES_REQUESTED** [review]
Things are much cleaner now, thanks @omdxp !

why do we have `as const` at [`mobile/src/screens/navigation.tsx:16`](https://github.com/dzcode-io/dzcode.io/pull/500/files#diff-120b1af0907f421f1477cc3adfacf2153df29f8696fae1233a67478a99b253e9R16) btw?

File: packages/ui-mobile/package.json (hunk 1 above)
Comment (ZibanPirate) [reply packages/ui-mobile/package.json:16]: don't forget to run `yarn` on the root after doing so

Comment (omdxp) [reply packages/ui-mobile/package.json:16]: Actually the `NavigationContainer` is still imported from `@react-navigation/native`, do you want to add it to the package as well ?

PS: `react-native-markdown-display` and `react-native-paper` are migrated in followup PRs

Also, you wrote `as const` line 😅, I'll remove it

Comment (omdxp) [reply packages/ui-mobile/package.json:16]: About navigation themes, they're migrated in the followup PR, so after merging this the `@react-navigation/native` and `react-native-paper` dependencies can be removed from `./mobile/package.json`

**Review by ZibanPirate — APPROVED** [review]

File: packages/ui-mobile/package.json (hunk 1 above)
Comment (ZibanPirate) [reply packages/ui-mobile/package.json:16]: i see, cool 👌🏽

**Pull request merged**